src/lesson/handlers.ts

```diff
@@ -1,42 +1,90 @@
-    });
+  getLessonByIdRequstValidator(async (req, res) => {
+    const { id } = req.params;
+    const lesson = await lessonsDal.getById(id).lean();
+
+    if (isNil(lesson)) {
+      throw new NotFoundError(`could not find lesson with id ${id}`);
+    }
+    res.send(lesson);
+  });
 
 export const createLesson = (
-    lessonsDal: LessonsDal,
-    videoSummeraizer: VideoSummeraizer
+  lessonsDal: LessonsDal,
+  videoSummeraizer: VideoSummeraizer
 ) =>
-    createLessonRequstValidator(async (req, res) => {
-        const { videoUrl } = req.body;
-        const videoId = extractVideoId(videoUrl);
-        const videoDetails = await getVideoDetails(videoId)
-        const summary = await videoSummeraizer.summerizeVideo(videoId);
-
-        const lesson = await lessonsDal.create({
-            owner: 'owner Mock', // TODO: use logged user after auth
-            sharedUsers: [],
-            summary,
-            title: videoDetails?.title ?? '',
-            videoUrl
-        });
-        res.status(StatusCodes.CREATED).json(lesson.toObject());
+  createLessonRequstValidator(async (req, res) => {
+    const { videoUrl } = req.body;
+    const videoId = extractVideoId(videoUrl);
+    const videoDetails = await getVideoDetails(videoId);
+    const summary = await videoSummeraizer.summerizeVideo(videoId);
+
+    const lesson = await lessonsDal.create({
+      owner: "owner Mock", // TODO: use logged user after auth
+      sharedUsers: [],
+      summary,
+      title: videoDetails?.title ?? "",
+      videoUrl,
+    });
+    res.status(StatusCodes.CREATED).json(lesson.toObject());
+  });
+
+export const getLessons =
+  (lessonsDal: LessonsDal) => async (req: any, res: any) => {
+    const lessons = await lessonsDal.getAll();
+    res.status(StatusCodes.OK).json(lessons);
+  };
+
+export const deleteLesson = (lessonsDal: LessonsDal) =>
+  deleteLessonRequstValidator(async (req, res) => {
+    const { id } = req.params;
+
+    const lesson = await lessonsDal.getById(id).lean();
+    if (isNil(lesson)) {
+      throw new NotFoundError(`Could not find lesson with id ${id}`);
+    }
+
+    await lessonsDal.deleteById(id);
+    res
+      .status(StatusCodes.OK)
+      .send({ message: `Lesson with id ${id} deleted successfully.` });
+  });
+
+export const updateLesson = (lessonsDal: LessonsDal) =>
+  updateLessonRequstValidator(async (req, res) => {
+    const { id } = req.params;
+    const {title, summary, videoUrl} = req.body;
+
+    const lesson = await lessonsDal.getById(id).lean();
+    if (isNil(lesson)) {
+      throw new NotFoundError(`Could not find lesson with id ${id}`);
+    }
+
+    const updatedLesson = await lessonsDal.updateById(id, {
+      title,
+      summary,
+      videoUrl,
     });
+
+    if (updatedLesson !== null) {
+      res.status(StatusCodes.OK).json(updatedLesson.toObject());
+    } else {
+      res
+        .status(StatusCodes.INTERNAL_SERVER_ERROR)
+        .json({ message: "Failed to update lesson" });
+    }
```

Comment:
but you already checked if the lesson exist, it is not supposed to be null.
my opinion is to remove the first getById check and throw here NotFound
```suggestion
    if (isNil(updatedLesson)) {
      throw new NotFoundError(`Could not find lesson with id ${id}`);
    }
    
    res.status(StatusCodes.OK).json(updatedLesson.toObject());
```